libexec/rtld-elf/rtld_c18n.h

```diff
@@ -118,9 +118,34 @@ struct stk_table {
 
 #define	COMPART_ID_MAX		index_to_cid((stk_table_index) { -1 })
 
+#include "rtld_c18n_machdep.h"
+
 struct tcb *c18n_allocate_tcb(struct tcb *);
 void c18n_free_tcb(void);
 
+/*
+ * When entering the RTLD without a trampoline (e.g., during lazy binding, TLS
+ * lookup, or stack resolution), a dummy trusted frame indicating that the
+ * current compartment is RTLD must be pushed.
+ */
+static inline struct trusted_frame *
+push_dummy_rtld_trusted_frame(struct trusted_frame *tf)
```

Comment:
The only case when this isn't get_trusted_stk() is when you first create the stack. Implicitly use get_trusted_stk(), and then upon creation you just need to set_trusted_stk() before calling this.